dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/logger/LogcatLogHandler.kt

```diff
@@ -92,11 +92,12 @@ internal class LogcatLogHandler(
         private const val MAX_TAG_LENGTH = 23
 
         private val ANONYMOUS_CLASS = Regex("(\\$\\d+)+$")
+
         // internal for testing
         internal val IGNORED_CLASS_NAMES = arrayOf(
             Logger::class.java.canonicalName,
             LogHandler::class.java.canonicalName,
-            LogHandler::class.java.canonicalName + "\$DefaultImpls",
+            LogHandler::class.java.canonicalName?.plus("\$DefaultImpls"),
```

Comment:
that is actually an interesting point: which class name is used in the stack trace elements: canonical (`getCanonicalName`) or just `Class#getName`? Apparently there is a difference between the 2 https://stackoverflow.com/questions/15202997/what-is-the-difference-between-canonical-name-simple-name-and-class-name-in-jav

Reply:
Yes, that's interesting. I actually don't know which version is being used in the stacktrace, I guess we could open a ticket for this if you want.

Reply:
if we assume that it is canonical name (which makes sense), then `LogHandler::class.java.canonicalName + "\$DefaultImpls"` will never match, because by looking at https://stackoverflow.com/questions/15202997/what-is-the-difference-between-canonical-name-simple-name-and-class-name-in-jav it seems canonical names can never have `$` symbol, it is replaced by `.` there. let's indeed open a ticket to check this topic.